programs/bench.c

```diff
@@ -536,33 +600,35 @@ static void BMK_benchCLevel(const void* srcBuffer, size_t benchedSize,
 
     for (l=cLevel; l <= cLevelLast; l++) {
         if (l==0) continue;  /* skip level 0 */
-        BMK_benchMem(srcBuffer, benchedSize,
-                     displayName, l,
-                     fileSizes, nbFiles,
-                     dictBuffer, dictBufferSize, compressionParams);
+        resultarray[l - cLevel] = BMK_benchMemCtxless(srcBuffer, benchedSize,
+                    displayName, l, fileSizes, nbFiles,
+                    dictBuffer, dictBufferSize, compressionParams, printable);
     }
+    returnval = BMK_average(resultarray, cLevelLast - cLevel + 1, NULL);
```

Comment:
Averaging the performance of multiple levels to generate a `BMK_result_t` return type is not good.
Such measure makes no sense.
If the only reason to do that is to retrieve a `BMK_result_t` for `paramgrill`, then consider skipping this stage.

There is still a need for a function that loops through multiple compression levels, but that's not for `paramgrill`.